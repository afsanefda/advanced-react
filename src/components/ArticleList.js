import React from 'react';
import Article from './Article';

const ArticleList = ({articles, store}) => {
  return (
    <div>
      {
        Object.values(articles).map(item =>
          <Article
            key={item.id}
            article={item}
            store={store}
          />
        )
      }
    </div>
  );
};

export default ArticleList;
