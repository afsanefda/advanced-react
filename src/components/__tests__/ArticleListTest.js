/* snapshot test of article list */
import React from 'react';
import ArticleList from '../ArticleList';
import renderer from 'react-test-renderer';

describe('snapshot test of article list', ()=> {
  let tree;
  /* mock props */
  const testProps = {
    articles: {
      a: { id: 'a', title: 'title a', date: 'date a', body: 'body a', authorId: 'a' },
      b: { id: 'b', title: 'title b', date: 'date b', body: 'body b', authorId: 'b' },
    },
    store: {
      lookupAuthor: jest.fn(()=> ({}))
    }
  };

  /* create tree before each it */
  beforeEach(() => {
    /* creating a test tree */
    tree = renderer.create(
      <ArticleList {...testProps} />
    ).toJSON();
  });

  it('should render correctly', () => {
    /* test tree should match with the main one */
    expect(tree).toMatchSnapshot();
  });

  it('should have correct children', () => {
    /* other checking */
    expect(tree.children.length).toBe(2);
  });
});
