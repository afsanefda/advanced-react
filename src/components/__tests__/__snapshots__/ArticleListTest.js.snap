// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of article list should render correctly 1`] = `
<div>
  <div
    className="sc-AxjAm buwshB"
  >
    <span
      className="sc-AxirZ cysuWk"
    >
      title a
    </span>
    <span
      className="sc-AxiKw dnkZXQ"
    >
      date a
    </span>
    <div
      className="sc-AxhCb jtmMmi"
    >
      <a>
        undefined undefined
      </a>
    </div>
    <span
      className="sc-AxhUy hDqVax"
    >
      body a
    </span>
  </div>
  <div
    className="sc-AxjAm buwshB"
  >
    <span
      className="sc-AxirZ cysuWk"
    >
      title b
    </span>
    <span
      className="sc-AxiKw dnkZXQ"
    >
      date b
    </span>
    <div
      className="sc-AxhCb jtmMmi"
    >
      <a>
        undefined undefined
      </a>
    </div>
    <span
      className="sc-AxhUy hDqVax"
    >
      body b
    </span>
  </div>
</div>
`;
